Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb

```diff
@@ -117,15 +140,63 @@ def process(queue)
           # Work-around: After returned by get_blob, the etag will contains quotes.
           new_etag = blob.properties[:etag]
           # ~ Work-around
-          blob, content = @azure_blob.get_blob(@container, blob_name, {:start_range=>start_index} )
-          
-          @codec.decode(content) do |event|
-            decorate(event)
-            queue << event
-          end # decode
+
+          blob, header = @azure_blob.get_blob(@container, blob_name, {:end_range => (@file_head_bytes-1) }) if header.nil? unless @file_head_bytes.nil? or @file_head_bytes <= 0
+
+          if start_index == 0
+            # Skip the header since it is already read.
+            start_index = @file_head_bytes
+          else
+            # Adjust the offset when it is other than first time, then read till the end of the file, including the tail.
+            start_index = start_index - @file_tail_bytes
+            start_index = 0 if start_index < 0
+          end
+
+          blob, content = @azure_blob.get_blob(@container, blob_name, {:start_range => start_index} )
+
+          # content will be used to calculate the new offset. Create a new variable for processed content.
+          processed_content = content
+
+          is_json_codec = @codec.is_a? LogStash::Codecs::JSON
+          if (is_json_codec)
+            skip = processed_content.index '{'
+            processed_content = processed_content[skip..-1] unless skip.nil?
+          end #if
+
+          if is_json_codec && (@break_json_down_policy != 'do_not_break')
+            @logger.debug("codec is json and policy is not do_not_break")
+            
+            @break_json_batch_count = 1 if break_json_batch_count <= 0
+            tail = processed_content[-@file_tail_bytes..-1]
+            while (!processed_content.nil? && processed_content.length > @file_tail_bytes)
+              json_event, processed_content = get_jsons(processed_content, @break_json_batch_count)
+              @logger.debug("Got json: ========================")
+              @logger.debug("#{json_event[0..50]}...#{json_event[-50..-1]}")
```

Comment:
why 0..50? I thought this is configurable.

Reply:
Are you asking json_even[0..50]...json_event[-50..-1]? That is just log output for debugging. I would like to see the start / end of the json event parsed, because if there's something wrong, it will like happen there near the beginning or ending.

Reply:
I misread your code earlier when I added the comment, so I take back what I said.
However, if your json_event is less than 50 characters long, would this throw an exception?

Reply:
Good concern on exception thrown. No, it won't thrown. It might output the same json twice.